### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by juan-apa**
## Features
- Adds docker (-compose) support to the project
- Adds docker instructions in readme

### CODE REVIEW

File: README.md
Comment (brunvez) [inline README.md:48]: Since `database.yml` is committed into the repo I think we can use env vars and not make the user configure stuff. Like `host: <%= ENV.fetch('DB_HOST') { 'localhost' } %>

**Review by brunvez — APPROVED** [review]

File: README.md
Comment (juan-apa) [reply README.md:48]: Actually if we move to dotenv, then I could pick the en variables from there to use them in here and pass them from docker to rails. I'll wait until that pr is made/merged and update accordingly. Wdyt?

**Review by santib — APPROVED** [review]

File: Dockerfile
Comment (JulianPasquale) [inline Dockerfile:35]: Can we use [cmd](https://docs.docker.com/engine/reference/builder/#cmd) here instead of `entrypoint`? So we can run this container overriding the command and for example run the tests or use a Rails console.

And we can also add a file with the entrypoint commands (for example in a new folder `docker/entrypoint.sh`):

```bash
#!/bin/bash
set -e

if [ -f /app/tmp/pids/server.pid ]; then
  rm /app/tmp/pids/server.pid
fi

exec "$@"
```

Finally in the Dockerfile, we will have
```dockerfile
ENTRYPOINT ["$APP_HOME/docker/docker-entrypoint.sh"]

CMD bundle exec rails server -b 0.0.0.0
```

File: Dockerfile
Comment (JulianPasquale) [inline Dockerfile:25]: What do you think about using `COPY` instead of `ADD`? As [Docker best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy) suggests

File: Dockerfile
Comment (juan-apa) [reply Dockerfile:35]: great idea! will do

File: Dockerfile
Comment (juan-apa) [reply Dockerfile:25]: Yep, makes sense

File: README.md
Comment (brunvez) [reply README.md:48]: +1 to that, still needs to be on the guide though right? To Docker or not to Docker

File: Dockerfile
Comment (JulianPasquale) [inline Dockerfile:1]: We are using Ruby 3.0 at main branch at this moment, so maybe we will need to change the image right?

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**
